changelog/duplicate-implementations-deprecation.dd

```diff
@@ -0,0 +1,26 @@
+Usage of identical functions in a single module
+
+A binary should never contain multiple versions of a symbol.
+Before this version, it was allowed to provide multiple implementation for a function,
+even if they result in the same mangling, which can result in undefined behaviour:
```

Comment:
Instead of saying it was "allowed", I would say that it was either "not detected", or similar.
Not a native speaker, but allowed could imply intent, while it was really a missing diagnostic.

My litmus test for this is: "Does this deprecation prevent me from doing something I was able to do before", and here's the answer clearly no.